### PR TITLE
OCP-1398 do not use resources

### PR DESCRIPTION
**Pull request by filosganga**
This avoids using the `Resources` in the Deduplication interface.

Also, we don't need to handle interruption and cancellation as if the effect is cancelled, we must no continue.

I have reworked the `Deduplication` interface to hide some internal details and expose only 3 methods:

- `tryProcess`: This will try to start processing. If another process is still in progress will wait until completed or expired
- `commitProcess`: This will complete the ongoing process (Why  not calling it `completeProcess`?)
- `protect`: This will make sure the wrapped effect happens only once.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
-# Comms-deduplication
+# Mnemosyne
```
Comment (ducanhkhu-ovo) [inline README.md:1]: why not `Moneta`? 😄

File: README.md [hunk 2]
```diff
@@ -22,23 +28,19 @@ The DynamoDb table has this structure:
...
 
-Each time a processor with a given `processorId` attempt to process a signal identified by `id`, it updates or writes on the table a record with `id`, `processorId`, `startedAt` and `expiresOn`. The `expiresOn` is allows to recover from a signal process that has never completed. Its value should be a little bit longher than time needed to process a signal.
+Each time a processor with a given `processorId` attempt to process a signal identified by `id`, it updates or writes on the table a record with `id`, `processorId`, `startedAt`. If the record with given `id` and `processorId` was already present, its value is returned to the library otherwise nothing is returned. After the process has runned successfully, the library mark it as completed by storing the `completedAt` and the `expiresOn` fields.
 
-If the record with given `id` and `processorId` was already present, its value is returned to the library otherwise nothing is returned.
+The `expiresOn` allows to clean up old data and re-run duplicate after some time.
 
-We can have these scenarios:
+When the librry attemp to start a process, these scenarios can happen:
```
Comment (ducanhkhu-ovo) [inline README.md:39]: `librry` typo

File: README.md [hunk 3]
```diff
@@ -14,6 +14,12 @@ It is able to work across multiple node with the same `processorId`. The persist
 - Strong consistency writes
 - Upsert with the previous record values returned
 
+## How to configure it
+
+A `processorId` need to be assigned to each instance of this library. It will univoque identify the processor. If two services have the same `processorId` it likely means they are two instance of the same service.
```
Comment (ducanhkhu-ovo) [inline README.md:19]: `univoque` typo, `instance` -> `instances`

File: README.md [hunk 4]
```diff
@@ -14,6 +14,12 @@ It is able to work across multiple node with the same `processorId`. The persist
...
+
+We need to know the max amount of time the process will take (`maxProcessingTime` in the config). Any process that take more than this amount of time will be considered death.
```
Comment (ducanhkhu-ovo) [inline README.md:21]: `death` -> `dead`

File: README.md [hunk 5]
```diff
@@ -22,23 +28,19 @@ The DynamoDb table has this structure:
...
-4) The signal has been attempted to be processed previously (`completedAt` is absent and `expiresOn` is in the past)
-
-In cases (1) and (4) the library allow the signal to be processed. In cases (2) and (3) the library does not allow the signal to be process again.
-
-When the signal process completes successfully, the record is updated with the `completedAt` time.
+3) The signal has timeout processing (`completedAt` is absent and `startedAt` + `processingTime` is in the past)
+4) The signal is stil being processing (`completedAt` is absent and `startedAt` + `processingTime` is in the future)
 
-As you can notice that `startedAt` end `expiresOn` change at any attempt to process the same signal, this is why we cannot have a DynamoDb TTL mechanism in place. This is a tradeoff to avoid using the conditional writes that are ore coslty.
+In cases (1) and (2) the library allow the signal to be processed. In cases (3) the library does not allow the signal to be processed again and (4) the library wait the process to either complete or timeout before taking any decision.
```
Comment (ducanhkhu-ovo) [inline README.md:46]: `allow` -> `allows`

File: src/it/scala/deduplication/DeduplicationSpec.scala [hunk 6]
```diff
@@ -14,150 +17,104 @@ import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
...
-          }
-          _ <- IO.sleep(5.seconds)
-          b <- ps.processing(id).map {
-            case ProcessStatus.NotStarted | ProcessStatus.Expired => "nonProcessed"
-            case _ => "processed"
-          }
-        } yield (a, b)
-      }
-      .unsafeRunSync()
+  it should "re-process the event if it failes the first time" in forAll {
```
Comment (ducanhkhu-ovo) [inline src/it/scala/deduplication/DeduplicationSpec.scala:66]: typo `failes`

File: README.md (hunk 1 above)
Comment (ra-ovo) [reply README.md:1]: The cat was walking on @filosganga's keyboard. Coin in Italian seems like a good name.

**Pull request merged**